edx_name_affirmation/tasks.py

```diff
@@ -43,58 +43,64 @@ def idv_update_verified_name_task(self, attempt_id, user_id, name_affirmation_st
     # want to grab all verified names for the same user and name combination, because
     # some of those records may already be associated with a different IDV attempt.
     verified_names = VerifiedName.objects.filter(
-        (Q(verification_attempt_id=attempt_id) | Q(verification_attempt_id__isnull=True))
+        (Q(platform_verification_attempt_id=attempt_id) | Q(platform_verification_attempt_id__isnull=True))
         & Q(user__id=user_id)
         & Q(verified_name=photo_id_name)
     ).order_by('-created')
+    verified_names_updated = False
     if verified_names:
         # if there are VerifiedName objects, we want to update existing entries
         # for each attempt with no attempt id (either proctoring or idv), update attempt id
         updated_for_attempt_id = verified_names.filter(
             proctored_exam_attempt_id=None,
-            verification_attempt_id=None
-        ).update(verification_attempt_id=attempt_id)
+            verification_attempt_id=None,
+            platform_verification_attempt_id=None
+        ).update(platform_verification_attempt_id=attempt_id)
 
         if updated_for_attempt_id:
+            verified_names_updated = True
             log.info(
-                'Updated VerifiedNames for user={user_id} to verification_attempt_id={attempt_id}'.format(
+                'Updated VerifiedNames for user={user_id} to platform_verification_attempt_id={attempt_id}'.format(
                     user_id=user_id,
                     attempt_id=attempt_id,
                 )
             )
 
         # then for all matching attempt ids, update the status
         verified_name_qs = verified_names.filter(
-            verification_attempt_id=attempt_id,
+            platform_verification_attempt_id=attempt_id,
+            verification_attempt_id=None,
             proctored_exam_attempt_id=None
         )
 
         # Individually update to ensure that post_save signals send
         for verified_name_obj in verified_name_qs:
             verified_name_obj.status = name_affirmation_status
             verified_name_obj.save()
+            verified_names_updated = True
 
         log.info(
-            'Updated VerifiedNames for user={user_id} with verification_attempt_id={attempt_id} to '
+            'Updated VerifiedNames for user={user_id} with platform_verification_attempt_id={attempt_id} to '
             'have status={status}'.format(
                 user_id=user_id,
                 attempt_id=attempt_id,
                 status=name_affirmation_status
             )
         )
-    else:
-        # otherwise if there are no entries, we want to create one.
+
+    # if there are no entries to update, we want to create one.
+    if not verified_names_updated:
```

Comment:
This actually fixes what appears to be an existing bug. If a user has a previously verified name using a different method (eg proctoring) and an new IDV creation event is emitted a new verified name record is not created. The event just gets dropped. From a user standpoint this doesn't really happen since the UI will post a blank name before that event but it still seemed like a gap.